src/estimation/include/iDynTree/Estimation/AttitudeEstimator.h

```diff
@@ -0,0 +1,142 @@
+        virtual bool updateFilterWithMeasurements(const iDynTree::LinearAccelerometerMeasurements& linAccMeas,
+                                                  const iDynTree::GyroscopeMeasurements& gyroMeas,
+                                                  const iDynTree::MagnetometerMeasurements& magMeas) = 0;
+
+        /**
+         * @brief Propagate the states and associated uncertainties through properly defined propagation functions
+         * The underlying implementation depends on the type of filter being implemented.
+         *
+         * @return true/false if successful/not
+         */
+        virtual bool propagateStates() = 0;
+
+        /**
+         * @brief Get orientation of the body with respect to inertial frame, in rotation matrix form
+         *
+         * @param[out] rot Rotation matrix
+         * @return true/false if successful/not
+         */
+        virtual bool getOrientationEstimateAsRotationMatrix(iDynTree::Rotation& rot) = 0;
+
+        /**
+         * @brief Get orientation of the body with respect to inertial frame, in quaternion form
+         *
+         * @note quaternion has the form (real, imaginary) and is normalized
+         *
+         * @param[out] q Quaternion
+         * @return true/false if successful/not
+         */
+        virtual bool getOrientationEstimateAsQuaternion(iDynTree::Quaternion& q) = 0;
+
+        /**
+         * @brief Get orientation of the body with respect to inertial frame, in Euler's RPY form
+         *
+         * @param[out] rpy 3D vector containing roll pitch yaw angles
+         * @return true/false if successful/not
+         */
+        virtual bool getOrientationEstimateAsRPY(iDynTree::RPY& rpy) = 0;
+
+        /**
+         * @brief Get dimension of the state vector
+         * @return size_t size of state vector
+         */
+        virtual size_t getInternalStateSize() const = 0;
+
+        /**
+         * @brief Get internal state of the estimator
+         *
+         * @param[out] stateBuffer Span object as reference of the container where state vector should be copied to
+         * @return true/false if successful/not
+         */
+        virtual bool getInternalState(iDynTree::Span<double> & stateBuffer) const = 0;
+
+        /**
+         * @brief Get initial internal state of the estimator
+         *
+         * @param[out] stateBuffer Span object as reference of the container where state vector should be copied to
+         * @return true/false if successful/not
+         */
+        virtual bool getInternalInitialState(iDynTree::Span<double> & stateBuffer) const = 0;
+
+        /**
+         * @brief set internal state of the estimator
+         *
+         * @param[in] stateBuffer Span object as reference of the container from which the internal state vector should be assigned
+         * @return true/false if successful/not
+         */
+        virtual bool setInternalState(iDynTree::Span<double> & stateBuffer) = 0;
+    };
```

Comment:
I think in practice a method that you will probably need for all filters will be something that returns an default initial internal state, but explicitly defining the initial ${}^A R_B$ state.